apple/internal/multi_arch_binary_support.bzl

```diff
@@ -44,23 +46,34 @@ def subtract_linking_contexts(owner, linking_contexts, avoid_dep_linking_context
     additional_inputs = []
     linkstamps = []
     avoid_library_set = _build_avoid_library_set(avoid_dep_linking_contexts)
+    linker_inputs_seen = dict()
```

Comment:
we could use `bazel_skylib` [sets](https://github.com/bazelbuild/bazel-skylib/blob/main/docs/new_sets_doc.md), not sure if thats better perf or not though

Reply:
an older implementation used them, but i decided to keep to the existing setup for ease of review. Bazel 8 and later versions of Bazel 7 also have a native [set](https://bazel.build/rules/lib/core/set) type now, which I’m looking forward to being able to use.

Reply:
the underlying implementation of skylib sets are very similar to this, fwiw, so I’d be surprised if there was a performance advantage

Reply:
Sounds good!